lib/cosmos/dart/script/dart_decom_server.rb

```diff
@@ -0,0 +1,201 @@
+require File.expand_path('../../config/environment', __FILE__)
+require 'dart_common'
+require 'dart_logging'
+
+# Rails Json screws up COSMOS handling of Nan, etc.
+require "active_support/core_ext/object/json"
+module ActiveSupport
+  module ToJsonWithActiveSupportEncoder # :nodoc:
+    def to_json(options = nil)
+      super(options)
+    end
+  end
+end
+
+class DartDecomServer
+  include DartCommon
+
+  def query(request)
+    request_start_time = Time.now
+    Cosmos::Logger.info("#{request_start_time.formatted}: QUERY: #{request}")
+
+    begin
+      start_time_sec = request['start_time_sec']
+      start_time_usec = request['start_time_usec']
+      end_time_sec = request['end_time_sec']
+      end_time_usec = request['end_time_usec']
+
+      start_time = nil
+      end_time = nil
+      start_time = Time.at(start_time_sec, start_time_usec) if start_time_sec and start_time_usec
+      end_time = Time.at(end_time_sec, end_time_usec) if end_time_sec and end_time_usec
+
+      item = request['item']
+      raise "Item \"#{item}\" invalid" if item.length != 3
+
+      reduction = request['reduction'].to_s.upcase
```

Comment:
I get that you didn't want to put the reduction smarts in the server but how does a client know which table to use? DART is the only thing that knows how much data is in a given query. If you're collecting data at 1 per minute vs 10Hz you're talking a 600x difference in data.

Reply:
Clients should know if they want reduced data or not.

Reply:
Sure but which reduced table should they use? That's the part I don't think they know. Perhaps an API to get data volumes or something?

Reply:
For the reduced data at least, you can assume 60 samples per hour, 24 samples per day, and 365 samples per year accordingly.   For the non-reduced you don't necessarily know, but in general that should only be used for queries of time less than an hour or so.   In general, the client needs to look at how much time is being requested and get the right data set.

Reply:
An API to get sample count isn't a bad idea

Reply:
Alright ✋